spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/Query.java

```diff
@@ -63,6 +63,14 @@
 	boolean count() default false;
 
 	/**
+	 * Returns whether the query defined should be executed as exists projection.
+	 *
+	 * @since 1.10
+	 * @return
+	 */
+	boolean exists() default false;
```

Comment:
That works for string-based queries quite well. I added a commit to derive count queries from a `long` return type.